src/main/java/edu/hw3/task2/Clusterizer.java

```
@@ -0,0 +1,41 @@
+package edu.hw3.task2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+public class Clusterizer {
+
+    private Clusterizer() {
+    }
+
+    public static List<String> clusterize(String input) {
+        List<String> clusters = new ArrayList<>();
+        Stack<Character> stack = new Stack<>();
+        StringBuilder currentCluster = new StringBuilder();
+
+        for (char ch : input.toCharArray()) {
+            if (ch == '(') {
+                currentCluster.append(ch);
+                stack.push(ch);
+            } else if (ch == ')') {
+                if (stack.isEmpty()) {
+                    return new ArrayList<>();
```

Лучше кидать исключение, иначе получаем отложенную ошибку. И не факт, что она случится, но при этом логика некорректна. Принцип fail fast.
Плюс результат не отличается от случая пустой строки.